hivemq-edge/src/test/java/com/hivemq/embedded/internal/EmbeddedHiveMQImplHttpTest.java

```diff
@@ -0,0 +1,119 @@
+    private int randomPort;
+    private int randomApiPort;
+
+    @Before
+    public void setUp() throws Exception {
+        data = tmp.newFolder("data");
+        license = tmp.newFolder("license");
+        extensions = tmp.newFolder("extensions");
+        conf = tmp.newFolder("conf");
+        randomPort = RandomPortGenerator.get();
+        randomApiPort = RandomPortGenerator.get();
+
+        final @NotNull String configXmlString = "" +
+                "<hivemq>\n" +
+                "    <mqtt-listeners>\n" +
+                "        <tcp-listener>\n" +
+                "            <port>" +
+                randomPort +
+                "</port>\n" +
+                "            <bind-address>0.0.0.0</bind-address>\n" +
+                "        </tcp-listener>\n" +
+                "    </mqtt-listeners>\n" +
+                "    <admin-api>\n" +
+                "        <enabled>true</enabled>\n" +
+                "        <listeners>\n" +
+                "            <http-listener>\n" +
+                "                <port>" +
+                randomApiPort +
+                "</port>\n" +
+                "                <bind-address>0.0.0.0</bind-address>\n" +
+                "            </http-listener>\n" +
+                "        </listeners>\n" +
+                "    </admin-api>\n" +
+                "</hivemq>";
+        FileUtils.write(new File(conf, "config.xml"), configXmlString, StandardCharsets.UTF_8);
+    }
+
+    @Test(timeout = 20000L)
+    public void embeddedHiveMQ_whenAdminApiEnabled_readsConfig() throws Exception {
+        try (final EmbeddedHiveMQImpl embeddedHiveMQ = new EmbeddedHiveMQImpl(conf, data, extensions, license)) {
+            embeddedHiveMQ.start().join();
+
+            final ConfigurationService configurationService = embeddedHiveMQ.bootstrapConfig();
+
+            final List<Listener> mqttListeners = configurationService.listenerConfiguration().getListeners();
+            assertEquals(1, mqttListeners.size(), "The listener count should be 1");
+            assertEquals(randomPort, mqttListeners.get(0).getPort(), "The MQTT port should match");
+
+            final List<ApiListener> apiListeners = configurationService.apiConfiguration().getListeners();
+            assertEquals(1, apiListeners.size(), "The API listener count should be 1");
+            assertEquals(randomApiPort, apiListeners.get(0).getPort(), "The Admin API port should match");
+
+            final HttpClient httpClient = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NEVER).build();
+            final HttpRequest httpRequest =
+                    HttpRequest.newBuilder().uri(URI.create("http://localhost:" + randomApiPort + "/")).GET().build();
+
+            final HttpResponse<String> httpResponse =
+                    httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            assertEquals(307, httpResponse.statusCode(), "The status code should be 307");
```

Comment:
Sorry for not pointing it out.
For asserts we stick to AssertJ.
You will still see other assert styles in older code but the general rule of thumb is to go with AssertJ for new stuff.